X2WOTCCommunityPromotionScreen/Src/X2WOTCCommunityPromotionScreen/Classes/CPS_UIArmory_PromotionHero.uc

```diff
@@ -1132,6 +1232,15 @@ function bool GetCustomAbilityCost(const XComGameState_Unit UnitState, const nam
 			return true;
 		}
 	}
+	for (i = 0; i < class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost.Length; i++)
+	{
+		if (class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost[i].ClassName == 'AnySoldierClass' && 
```

Comment:
I'm OK with this, but note that I was fine with no class name on the basis an entry could look like this:
```
+ClassCustomAbilityCost= (AbilityName=Magnum_LW, AbilityCost=5) 
```
i.e. with no soldier class specified.

Reply:
Also, is a second loop really necessary? Could just expand the condition in the loop above:
```
if (class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost[i].AbilityName == AbilityName &&
    (class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost[i].ClassName == ClassTemplate.DataName ||
     class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost[i].ClassName == 'AnySoldierClass'))
```
At which point it might be better to assign `class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost[i]` to a local variable for clarity.

Reply:
Having two separate loops will ensure the function cycles through the array in full, looking for config line specific to this ability and this soldier class. And uses a "any soldier class" config only if there is no line like that. 

> it might be better to assign class'NPSBDP_UIArmory_PromotionHero'.default.ClassCustomAbilityCost[i] to a local variable for clarity.

Agreed.

>  I was fine with no class name

Let's go that way, because it's easier to remember that if you want the entry to apply to all classes then you just keep the entry empty, as opposed to trying to remember if it was "AnyClass" or "AnySoldierClass" or "AnySoldier", etc.

That said, there might be a use case for "AnyResistanceHero" and "AnyNonResistanceHero" entries.

Reply:
> Having two separate loops will ensure the function cycles through the array in full, looking for config line specific to this ability and this soldier class.

Ah, good point. Ensures that class specific entries take precedence over the general ones.